File: api/health.js
// Health check endpoint for Harper's Bazaar crawler
export default async function handler(req, res) {
  res.setHeader('Access-Control-Allow-Origin', '*')
  res.setHeader('Access-Control-Allow-Methods', 'GET, OPTIONS')
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type')

  if (req.method === 'OPTIONS') {
    return res.status(200).end()
  }

  if (req.method !== 'GET') {
    return res.status(405).json({ message: 'Method not allowed' })
  }

  res.status(200).json({
    status: 'healthy',
    service: 'harpers_bazaar_crawler',
    timestamp: new Date().toISOString(),
    version: '1.0.0',
    endpoints: {
      trending: '/api/trends/latest',
      featured: '/api/trends/featured',
      health: '/api/health'
    }
  })
}

File: api/health.py
# Vercel Serverless Function for Health Check
from http.server import BaseHTTPRequestHandler
import json
from datetime import datetime

class handler(BaseHTTPRequestHandler):
    def do_GET(self):
        # Set CORS headers
        self.send_response(200)
        self.send_header('Content-Type', 'application/json')
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'GET, POST, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type')
        self.end_headers()
        
        # Return health status
        response = {
            "status": "healthy",
            "service": "Looklyy API",
            "version": "1.0.0",
            "timestamp": datetime.now().isoformat(),
            "platform": "Vercel Serverless"
        }
        
        self.wfile.write(json.dumps(response).encode())
    
    def do_OPTIONS(self):
        # Handle CORS preflight
        self.send_response(200)
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'GET, POST, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type')
        self.end_headers()
